engine/src/main/java/org/hibernate/search/engine/backend/document/converter/ToIndexIdValueConverter.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.backend.document.converter;
+
+import org.hibernate.search.engine.backend.document.converter.runtime.ToIndexFieldValueConvertContext;
+import org.hibernate.search.engine.backend.document.converter.runtime.ToIndexIdValueConvertContext;
+import org.hibernate.search.engine.backend.document.converter.runtime.ToIndexIdValueConvertContextExtension;
+
+/**
+ * A converter from a source value to a target value that should be indexed.
+ *
+ * @param <V> The type of source values.
+ */
+public interface ToIndexIdValueConverter<I> {
```

Comment:
I know I suggested it, but on second thought, this name seems confusing: it looks like this "id" we are talking about identifies an index, and not a document. We could use "ToDocumentIdValueConverter", but then it would not be consistent with "ToIndexFieldValueConverter"...